experiment/resultstore/convert.go

```diff
@@ -231,3 +246,69 @@ func convert(project, details string, url gcs.Path, result downloadResult) (resu
 
 	return inv, target, test
 }
+
+func startedReposToProperties(gitRepos map[string]string) []resultstore.Property {
+	var properties []resultstore.Property
+
+	knownOrg := make(map[string]bool)
+	knownBranch := make(map[string]bool)
+	for repo, branch := range gitRepos {
+		orgRepo := strings.SplitN(repo, "/", 2)
+		org := orgRepo[0]
+		repoName := ""
+		if len(orgRepo) == 2 {
+			repoName = orgRepo[1]
+		} else {
+			repoName = org
+		}
+
+		if _, ok := knownOrg[org]; !ok {
+			knownOrg[org] = true
+			orgName := resultstore.Property{
+				Key:   "Org",
+				Value: org,
+			}
+			properties = append(properties, orgName)
+		}
+
+		if _, ok := knownBranch[branch]; !ok {
+			knownBranch[branch] = true
+			branchName := resultstore.Property{
+				Key:   "Branch",
+				Value: branch,
+			}
+			properties = append(properties, branchName)
+		}
+
+		repos := []resultstore.Property{
+			{
+				Key:   "Repo",
+				Value: repoName,
+			},
+			{
+				Key:   "Repo",
+				Value: repo,
+			},
+			{
+				Key:   "Repo",
+				Value: fmt.Sprintf("%s:%s", repo, branch),
+			},
+		}
+		properties = append(properties, repos...)
+	}
+	return properties
+}
+
+// prowJobName returns the prow job name parsed from the GCS bucket.
+// If parsing fails, it returns an empty string.
+// TODO: use prowjob.json when PR 15732 is done.
+func prowJobName(url string) string {
+	url = strings.TrimSpace(url)
+	url = strings.TrimPrefix(url, "gs://")
+	url = strings.TrimSuffix(url, "/")
+	split := strings.Split(url, "/")
+	if len(split) <= 2 {
+		return ""
+	}
+	return split[len(split)-2]
```

Comment:
Can't this be done more easily and correctly with a regex?

Reply:
This approach is easier to reason and is no more computationally expensive than regex. Also, waiting for PR 15785 and this will go away soon.